proposals/0000-unicode-for-string-processing.md

```diff
@@ -0,0 +1,1321 @@
+  /// by default.
+  ///
+  /// You can call this method to change the default repetition behavior for
+  /// quantifier operators in regex syntax and `RegexBuilder` quantifier
+  /// methods. For example, in the following example, both regexes use
+  /// possessive quantification when matching a quotation surround by `"`
+  /// quote marks:
+  ///
+  ///     let regex1 = /"[^"]*"/.defaultRepetitionBehavior(.possessive)
+  ///
+  ///     let quoteMark = "\""
+  ///     let regex2 = Regex {
+  ///         quoteMark
+  ///         ZeroOrMore(.noneOf(quoteMark))
+  ///         quoteMark
+  ///     }.defaultRepetitionBehavior(.possessive)
+  ///
+  /// This setting only changes the default behavior of quantifiers, and does
+  /// not affect regex syntax operators with an explicit behavior indicator,
+  /// such as `*?` or `++`. Likewise, calls to quantifier methods such as
+  /// `OneOrMore` always use the explicit `behavior`, when given.
+  ///
+  /// - Parameter behavior: The default behavior to use for quantifiers.
+  public func defaultRepetitionBehavior(_ behavior: RegexRepetitionBehavior) -> Regex<RegexOutput>
+}
+
+public struct RegexRepetitionBehavior {
+  /// Match as much of the input string as possible, backtracking when
+  /// necessary.
+  public static var eager: RegexRepetitionBehavior { get }
+
+  /// Match as little of the input string as possible, expanding the matched
+  /// region as necessary to complete a match.
+  public static var reluctant: RegexRepetitionBehavior { get }
+
+  /// Match as much of the input string as possible, performing no backtracking.
+  public static var possessive: RegexRepetitionBehavior { get }
+}
+```
+
+The `RegexBuilder` quantifier APIs include a `nil`-defaulted optional `behavior` parameter. When you pass `nil`, the quantifier uses the default behavior as set by this option. If an explicit behavior is passed, that behavior is used regardless of the default.
+
+```swift
+// Example `OneOrMore` initializer
+extension OneOrMore {
+    public init<W, C0, Component: RegexComponent>(
+    _ behavior: RegexRepetitionBehavior? = nil,
+    @RegexComponentBuilder _ component: () -> Component
+  ) where Output == (Substring, C0), Component.Output == (W, C0)
+}
+```
+
+#### Single line mode (`.` matches newlines)
+
+The "any" metacharacter (`.`) matches any character in a string *except* newlines by default. With the `s` option enabled, `.` matches any character including newlines.
+
+```swift
+let str = """
+    <<This string
+    uses double-angle-brackets
+    to group text.>>
+    """
+    
+str.firstMatch(of: /<<.+>>/)        // nil
+str.firstMatch(of: /(?s)<<.+>>/)    // "This string\nuses double-angle-brackets\nto group text."
+```
+
+This option applies only to `.` used in regex syntax and does _not_ affect the behavior of `CharacterClass.any`, which always matches any character or Unicode scalar. To get the default `.` behavior when using `RegexBuilder` syntax, use `CharacterClass.anyNonNewline`.
```

Comment:
Does this argue there should be a `dot` character class that _does_ obey this mode? One of the goals of `CharacterClass` is to give an API for what can be expressed in a literal.